frontend/src/components/ActivityList.vue

```diff
@@ -182,27 +196,31 @@ export default {
     },
   },
   methods: {
-    fetchItems() {
+    async fetchItems() {
+      this.isBusy = true;
+      this.items = [];
       let filter;
       if (this.filter && this.filterValue) {
         filter = { name: this.filter.value, value: this.filterValue.value };
       }
-      activitiesService
-        .listActivities(this.tasks, this.activities, filter)
-        .then((result) => {
-          if (Object.prototype.hasOwnProperty.call(result, "data")) {
-            this.isBusy = false;
-            this.items = result.data;
-          }
-        })
-        .catch((error) => {
-          this.isBusy = false;
-          if (error.response.status === 404) {
-            this.items = [];
-          } else {
-            EventBus.$emit("error", this.$axiosErrorMessage(error));
-          }
-        });
+      const params = PageOptions.toUrlSearchParams(this.options);
+      try {
+        const response = await activitiesService.listActivities(
+          this.tasks,
+          this.activities,
+          filter,
+          params
+        );
+        if (response.status === 200) {
```

Comment:
These 200 checks redundant. The axios default is to treat everything that is status 2xx as success, 3xx redirects are followed, and everything 4xx as an error. 

see https://github.com/axios/axios#request-config and the example:

```javascript
  validateStatus: function (status) {
    return status >= 200 && status < 300; // default
  },
```

Reply:
ahh cool, so i can just discard the if statement and it should work similar?

I followed your implementation in [PR #730](https://github.com/hyperledger-labs/business-partner-agent/pull/730), so i can remove the redundant 200 checks in every ***List.vue file i guess?

Reply:
as long as there is a try/catch block yes. the only exception being if you would want react on different status codes, but we never do that.